canopy_waf_mod.F90

```diff
@@ -57,7 +57,7 @@ SUBROUTINE CANOPY_FLAMEH( FLAMEH_OPT, FLAMEH_SET, DX, MODRES, &
             MIDFLAMEPOINT = 2    !do not put at z = 0, but rather in second layer
         else
             flamelays     = floor(FLAMEH/MODRES)
-            MIDFLAMEPOINT = max((flamelays/2),2)
+            MIDFLAMEPOINT = max((flamelays/2),2) + 1 !Add 1st level at ground (z=0)
```

Comment:
@zmoon I am only simply adding 1 additional layer to account for the ground surface at z = 0.  You can see, when FLAMEH = 2.0 m, the flamelays = 4 and MIDFLAMEPOINT = 2, which gives a height of z = 0.5 m (not correct), and adding one more layer gives z = 1.0 m, which is correct.

Reply:
Maybe I need to think about it more here..

Reply:
but if res = 0.5 m and 0.5 <= flameh < 1 m, flamelays = 1 -> midflamepoint = 2  + 1 => 1 m, which is greater than the flameh

Reply:
As I can see how it would break things if say MODRES =  1.0 or greater.    Need to change this then to somehow better account for these situations.

Reply:
Right, we need someway to get the correct midflamepoint, and to add in the additional z = 0 m first level.  Is it simply midflamepoint = floor(FLAMEH/2) + 1

Reply:
Maybe something like?   midflamepoint = floor[(flameh/2)/modres] + 1  ?

Reply:
Or, rather just:  (flameh/2)/modres  ?

Reply:
> Actually, maybe then we put it here then? flamelays = floor(FLAMEH/MODRES) + 1

Yeah I thought about that, maybe that is the way to go. This is confusing stuff, would be nice if we could extract and unit test it at some point haha.

Reply:
@zmoon OK, I fixed it here in the next [commit](https://github.com/noaa-oar-arl/canopy-app/pull/34/commits/b8193eb093c4a08d5081087025957dfa379c3990).

Reply:
@zmoon Once Hopper gets it together, I need to remove the "[ceiling](https://github.com/noaa-oar-arl/canopy-app/pull/34/commits/b8193eb093c4a08d5081087025957dfa379c3990#diff-2a4246c7a01a1e386f28e722b4300df1c0bb4e2135957ee6edcd3ac8b19bd828R60)" I added and I think this formulation works.  I also think this comes down to modres chosen too, where if the modres becomes >> flameh then some error is inherently incurred because the model is not well resolving the flame height region.   I will fix this in next commit and could add a note in the README explanation of modres.

Reply:
@zmoon I changed the midflamepoint calculation [here](https://github.com/noaa-oar-arl/canopy-app/pull/34/commits/e0ee1deb623e65363bd4497c8fbb7717e0b174bd).